### ISSUE TITLE
Suggestion: Form builder

### ISSUE THREAD

**Issue by FelixINX** [LaminSanneh/Octobercms-Flexi-Contact-Plugin#26]
Hi,

Here is the roadmap for the Form Builder!
- [ ] Multiple recipients [DevINX/oc-contactus#2](https://github.com/DevINX/oc-contactus/issues/2)
- [ ] Custom messages (#24)
- [ ] Custom inputs
- [ ] Custom inputs order
- [ ] Discontinuity of DevINX.ContactUs
- [ ] Rainlab.Translate compatibility
- [ ] Multiple forms
- [ ] Support for file-upload (#17)

I am open to suggestion! I will post update here.

Due to summer vacation, I will not be able to code during one month. However, I wil try to be avaible as much as I can.

**Comment by FelixINX** [comment]
Hi,

I need your opinion, I will not be able to extend the current Flexi Contact because of how it is created.  There are two options : 
1. Change from a Settings model to a Controller (I'm not very familiar with a controller without a List behavior)
2. Or, the best option for me, a complete rewrite of the plugin, with multiple forms. This will allow to have a better controller, with a list, and it will be more easier to extend.

What do you prefer @LaminSanneh @worldofsmarty?

Thanks!

**Comment by LaminSanneh** [comment]
@FelixINX , the rewrite is not a bad idea but then it will be another plugin as there really is no use in extending the current one if we will find it hard to incorporate the features above. We can call it pro, maybe even charge for it(just a though, but then it will have to be made private).

I vote for a rewrite and keeping the current one as a lite version.

**Comment by FelixINX** [comment]
This is a good idea. I will not need to make another plugin. I will give
you access to a Github repo soon containing the work I have done as of now.
Fell free to use it how you want.

Le mercredi 13 juillet 2016, Lamin Sanneh notifications@github.com a
écrit :

> @FelixINX https://github.com/FelixINX , the rewrite is not a bad idea
> but then it will be another plugin as there really is no use in extending
> the current one if we will find it hard to incorporate the features above.
> We can call it pro, maybe even charge for it(just a though, but then it
> will have to be made private).
> 
> I vote for a rewrite and keeping the current one as a lite version.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> https://github.com/LaminSanneh/Octobercms-Flexi-Contact-Plugin/issues/26#issuecomment-232343618,
> or mute the thread
> https://github.com/notifications/unsubscribe/AHo1xO_vbM0Z43SMfKPEiFiAqjh_J4GYks5qVN2hgaJpZM4JKbnX
> .

## 

Félix Desjardins

**Comment by LaminSanneh** [comment]
The repo you will give me access, is it a new one you just started with roadmap above?

**Comment by FelixINX** [comment]
Yes, not finished. The DB is finished for the current need of the plugin,
and I am working on the controller, backend administration. But as I say
previously, I will not be home for a month, and I will be unable to code. I
will try to be connected to Internet as much as I can for any questions.

Le mercredi 13 juillet 2016, Lamin Sanneh notifications@github.com a
écrit :

> The repo you will give me access, is it a new one you just started with
> roadmap above?
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> https://github.com/LaminSanneh/Octobercms-Flexi-Contact-Plugin/issues/26#issuecomment-232344541,
> or mute the thread
> https://github.com/notifications/unsubscribe/AHo1xEZejzE3GaI2THlojUk7vIKcfeg5ks5qVN6VgaJpZM4JKbnX
> .

## 

Félix Desjardins

**Comment by LaminSanneh** [comment]
Ok, please fill me in on the plugin. is it possible to publish plugin in october for multiple authors?

**Comment by FelixINX** [comment]
I dont think we can. I will look. But for sure, this is not ready for
production :-)

Le mercredi 13 juillet 2016, Lamin Sanneh notifications@github.com a
écrit :

> Ok, please fill me in on the plugin. is it possible to publish plugin in
> october for multiple authors?
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> https://github.com/LaminSanneh/Octobercms-Flexi-Contact-Plugin/issues/26#issuecomment-232350184,
> or mute the thread
> https://github.com/notifications/unsubscribe/AHo1xOhnAFuqz7moFn0MfRAgvvo9E0cAks5qVOQ8gaJpZM4JKbnX
> .

## 

Félix Desjardins

**Comment by LaminSanneh** [comment]
Yeah that's fine so whats the plan for publishing?

**Comment by LaminSanneh** [comment]
Plenty of time to code it out, not in a hurry.

**Comment by FelixINX** [comment]
I will do my best to finish the controller for tomorrow. The logic (sending email, recaptcha) should be very similar as what is is right now.
I think that by end of summer (end August/September), the plugin would be ready for a stable use.

I will add some notes for you on the README.

**Comment by LaminSanneh** [comment]
Ok thats fine. What I am asking is under whos name will it be released?

**Comment by FelixINX** [comment]
You decide. It can be your, mine or another. I do not mine about this. Do you have private repo on your account? I could push the code there.

**Comment by LaminSanneh** [comment]
No, not on github, only on bitbucket but we can take care of that later.

**Comment by worldofsmarty** [comment]
Sorry for the late reply, I would rather suggest to go for this plugin as a free version. As there is another plugin available in the October Market with same features. And this could be another plugin in competition with that. https://octobercms.com/plugin/rafie-contactform. This is just a suggestion. If you need any inputs or help in developing this plugin. I would be happy to help.

**Comment by FelixINX** [comment]
Hi @worldofsmarty !
I think that a free plugin is good idea because it allow people to better use OctoberCMS, then more people in the community!

Also, for any October developer, I have a question : 
I want to give the user the choice of the mail template. How can I make a relation with Mail Template? It should be `hasOne`.

**Comment by worldofsmarty** [comment]
Well we can give the user the choice of choosing the template from the inspector. As i had purchased the contact plugin https://octobercms.com/plugin/fabriciorabelo-contact . It had that interface and i feel it will be easy for user to choose which template to use while adding the component to the web page. I have attached the screenshot of the plugin as well. Even i can share the plugin source if you need as i have the whole repo of the plugin. Just let me know.
![screen shot 2016-07-14 at 9 29 29 am](https://cloud.githubusercontent.com/assets/5152491/16827631/381a4e3c-49a6-11e6-8874-df945584af52.png)

**Comment by FelixINX** [comment]
I don't think it should be inside the component, because all the others options will be in the backend.

I don't need the source code, I will make a post on the forum to find how to do this

**Comment by LaminSanneh** [comment]
@worldofsmarty thanks for the offer but i believe sharing code for a sold product is sort of inappropriate. We should have a balance between what should be in the component page and what should be an option.

A good start would be to have defaults for templates, message body e.t.c, and have them overrideable when inserting the component. How's that sound?

**Comment by FelixINX** [comment]
Great! This is what I wanted to do first. It will be as the plugin now, one
template per form, will inputs preloaded. But I am not sure if we can
change the template based on the form inputs. Have you any idea?

Le jeudi 14 juillet 2016, Lamin Sanneh notifications@github.com a écrit :

> @worldofsmarty https://github.com/worldofsmarty thanks for the offer
> but i believe sharing code for a sold product is sort of inappropriate. We
> should have a balance between what should be in the component page and what
> should be an option.
> 
> A good start would be to have defaults for templates, message body e.t.c,
> and have them overrideable when inserting the component. How's that sound?
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> https://github.com/LaminSanneh/Octobercms-Flexi-Contact-Plugin/issues/26#issuecomment-232637843,
> or mute the thread
> https://github.com/notifications/unsubscribe/AHo1xBdD8uJ8Q69YhIg-G2-mPOMmeCoPks5qVhvjgaJpZM4JKbnX
> .

## 

Félix Desjardins

**Comment by LaminSanneh** [comment]
Yeah I believe one template per form. One question, isn't templates already part of a form when creating the form. 

I might not be understanding the proposal enough so enlighten me how we could have many templates per form or vice-versa.

**Comment by FelixINX** [comment]
Here is something a created for you :smile: 

![http://imgh.us/flexiplus.svg](http://imgh.us/flexiplus.svg)

**Comment by LaminSanneh** [comment]
Hey thanks for that. Ok, so by my understanding, the template will be a one-to-one with the form and would be generated from the inputs we add to the form during creation.
